src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -591,6 +622,101 @@ protected void deleteNetworks(Azure azureService, List<com.microsoft.azure.manag
+                                                        .withType(SCRIPT_EXTENSION_TYPE_LINUX)
+                                                        .withVersion(SCRIPT_EXTENSION_VERSION_LINUX)
+                                                        .withMinorVersionAutoUpgrade()
+                                                        .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, script)
+                                                        .attach()
+                                                        .applyAsync(null);
+            try {
+                vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+            } catch (InterruptedException e) {
+                log.error("An interruption occurred while installing extension and executing script to a Linux VM.");
+                Thread.currentThread().interrupt();
+                e.printStackTrace();
+            } catch (ExecutionException e) {
+                log.error("An error occurred when installing extension and executing script to a Linux VM.");
+                e.printStackTrace();
+            } catch (TimeoutException e) {
+                log.info("Timeout reached when installing extension and executing script to a Linux VM.");
+            }
+
+        } else if (vm.osType().equals(OperatingSystemTypes.WINDOWS)) {
+            AbstractFuture<VirtualMachine> vmfuture = vm.update()
+                                                        .defineNewExtension(createUniqueScriptName(vm.name()))
+                                                        .withPublisher(SCRIPT_EXTENSION_PUBLISHER_WINDOWS)
+                                                        .withType(SCRIPT_EXTENSION_TYPE_WINDOWS)
+                                                        .withVersion(SCRIPT_EXTENSION_VERSION_WINDOWS)
+                                                        .withMinorVersionAutoUpgrade()
+                                                        .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, script)
+                                                        .attach()
+                                                        .applyAsync(null);
+            try {
+                vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+            } catch (InterruptedException e) {
+                log.error("An interruption occurred while installing extension and executing script to a Windows VM.");
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException e) {
+                log.info("An error occurred when installing extension and executing script to a Windows VM.");
+                e.printStackTrace();
+            } catch (TimeoutException e) {
+                log.info("Timeout reached when installing extension and executing script to a Windows VM.");
+            }
+        } else {
+            throw new RuntimeException(unsupportedOperatingSystemError(vm.osType().toString()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)